less_users/tests/test_user_challenge.py

```diff
@@ -1,18 +1,119 @@
-from less_users.models import UserChallenge
+import pytest
 
+from freezegun import freeze_time
+from datetime import date
 
-def test_user_challenge_correct_creation(user, challenge_1_day, create_user_challenge):
-    user_challenge = create_user_challenge
+from less_users.models import UserChallenge, Log
+
+
+def test_user_challenge_correct_creation(
+    user, challenge_1_day, create_user_challenge_day
+):
+    user_challenge = create_user_challenge_day
 
     assert user_challenge in UserChallenge.objects.all()
     assert user_challenge.user == user
     assert user_challenge.challenge == challenge_1_day
 
 
-def test_user_challenge_deactivation(create_user_challenge):
-    user_challenge = create_user_challenge
-    user_challenge.is_active = False
+@pytest.mark.django_db
+@freeze_time("2023-01-01")
+def test_user_challenge_deactivation_when_duration_passed(user, challenge_1_day):
+    new_challenge = UserChallenge(
+        user=user,
+        challenge=challenge_1_day,
+    )
+    new_challenge.save()
+    new_challenge.days_left(todays_date=date(2023, 1, 5))
+    new_challenge.check_if_active()
+
+    assert not new_challenge.is_active
+
+
+def test_user_challenge_done(challenge_1_day):
+    pass
+
+
+@pytest.mark.django_db
+@freeze_time("2022-12-01")
+def test_user_challenge_model_days_left_3(user, challenge_3_month):
+    new_challenge = UserChallenge(user=user, challenge=challenge_3_month)
+    new_challenge.save()
+
+    todays_date = date(2022, 12, 28)
+    assert new_challenge.days_left(todays_date=todays_date) == 3
+
+
+@pytest.mark.django_db
+@freeze_time("2022-12-31")
+def test_user_challenge_model_days_left_12(user, challenge_3_month):
+    new_challenge = UserChallenge(user=user, challenge=challenge_3_month)
+    new_challenge.save()
+
+    todays_date = date(2023, 1, 18)
+    assert new_challenge.days_left(todays_date=todays_date) == 12
+
+
+@pytest.mark.django_db
+@freeze_time("2023-01-29")
+def test_user_challenge_model_days_left_less_than_0(user, challenge_3_month):
+    new_challenge = UserChallenge(user=user, challenge=challenge_3_month)
+    new_challenge.save()
+
+    todays_date = date(2023, 3, 1)
+    assert new_challenge.days_left(todays_date=todays_date) == -1
+
+
+@pytest.mark.django_db
+def test_user_challenge_model_days_left_30(user, challenge_3_month):
+    new_challenge = UserChallenge(user=user, challenge=challenge_3_month)
+    new_challenge.save()
+
+    assert new_challenge.days_left() == 30
```

Comment:
https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
wiedziałam, że gdzieś jest na to sposób, żeby przypisać parametry testu. Dzięki za podpowiadajkę!